src/scout_apm/django/middleware.py

```diff
@@ -20,14 +22,34 @@ def __init__(self, get_response):
         self.get_response = get_response
 
     def __call__(self, request):
-        operation = "Middleware"
+        tr = TrackedRequest.instance()
+
+        self._track_request_queue_time(request, tr)
+        tr.start_span(operation="Middleware")
 
-        TrackedRequest.instance().start_span(operation=operation)
         try:
-            response = self.get_response(request)
+            return self.get_response(request)
         finally:
             TrackedRequest.instance().stop_span()
-        return response
+
+    _queue_time_re = re.compile(r'(t=|\.)')
+
+    def _track_request_queue_time(self, request, tracked_request):
+        header_value = (
+            request.META.get("HTTP_X_QUEUE_START") or
+            request.META.get("HTTP_X_REQUEST_START")
+        )
+        if not header_value:
+            return
+
+        raw_start = self._queue_time_re.sub('', header_value)
+        parsed_start = datetime.utcfromtimestamp(float(raw_start[0:10] + '.' + raw_start[10:13]))
+
+        tr.start_span(
+            operation="QueueTime/Request",
+            start_time=parsed_start,
```

Comment:
This may cause issues if this time is before the request start time. Unsure how that works in the core agent, needs testing.

Reply:
You're right. I'll be testing on my demo app before adding unit tests so will let you know. Also the ruby agent has a guard against the start time being in the future which I need to add here.